delegator.py

```diff
@@ -151,6 +155,11 @@ def kill(self):
 
     def block(self):
         """Blocks until process is complete."""
+        if self._uses_subprocess:
+            # consume stdout and stderr
+            stdout, stderr = self.subprocess.communicate()
+            self.__out = stdout
+            self.__err = stderr
         self.subprocess.wait()
```

Comment:
Is `wait()` still necessary here? The [documentation](https://docs.python.org/2/library/subprocess.html#subprocess.Popen.communicate) for `communicate()` seems to suggest it runs until the subprocess terminates.